design/nfs/nfs-quota.md

```diff
@@ -0,0 +1,120 @@
+# NFS Quota
+
+## Background
+
+Currently, when the user creates NFS PersistentVolumes from an NFS Rook share/export via PersistentVolumeClaim, the provisioner does not provide the specific capacity as requested. For example the users create NFS PersistentVolumes via PersistentVolumeClaim as following:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: rook-nfs-pv-claim
+spec:
+  storageClassName: "rook-nfs-share"
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Mi
+```
+
+The client still can use the higher capacity than `1mi` as requested.
+
+This proposal is to add features which the Rook NFS Provisioner will provide the specific capacity as requested from `.spec.resources.requests.storage` field in PersistentVolumeClaim.
+
+## Implementation
+
+The implementation will be use `Project Quota` on xfs filesystem. When the users need to use the quota feature they should use xfs filesystem with `prjquota/pquota` mount options for underlying volume. Users can specify filesystem type and mount options through StorageClass that will be used for underlying volume. For example:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard-xfs
+parameters:
+  fsType: xfs
```

Comment:
just looking at this storageClass, i was thinking that saying `xfs` implied it would result in an underlying volume that it tied to a single host node.  But many types of distributed storage providers (e.g. google PD or amazon AWS) support XFS too, so using XFS in the underlying volume does not mean that the NFS server is tied to a single host. Correct?

is that worth adding a note about here (that many distributed volume types support XFS)?